cmd/dependabot/internal/cmd/update.go

```diff
@@ -156,6 +85,86 @@ var updateCmd = &cobra.Command{
 	},
 }
 
+func extractInput(cmd *cobra.Command) (*model.Input, []byte, error) {
+	var input *model.Input
+	var inputRaw []byte
+
+	if file != "" {
+		if len(cmd.Flags().Args()) > 0 {
+			return nil, nil, errors.New("cannot use file and arguments together")
+		}
+		var err error
+		input, inputRaw, err = readInputFile(file)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	if len(cmd.Flags().Args()) > 0 {
+		if len(cmd.Flags().Args()) != 2 {
+			return nil, nil, errors.New("requires a package manager and repo argument")
+		}
+
+		packageManager = cmd.Flags().Args()[0]
+		if packageManager == "" {
+			return nil, nil, errors.New("requires a package manager argument")
+		}
+
+		repo = cmd.Flags().Args()[1]
+		if repo == "" {
+			return nil, nil, errors.New("requires a repo argument")
+		}
+
+		input = &model.Input{
+			Job: model.Job{
+				PackageManager: packageManager,
+				AllowedUpdates: []model.Allowed{{
+					UpdateType: "all",
+				}},
+				DependencyGroups:           nil,
+				Dependencies:               nil,
+				ExistingPullRequests:       [][]model.ExistingPR{},
+				IgnoreConditions:           []model.Condition{},
+				LockfileOnly:               false,
+				RequirementsUpdateStrategy: nil,
+				SecurityAdvisories:         []model.Advisory{},
+				SecurityUpdatesOnly:        false,
+				Source: model.Source{
+					Provider:    provider,
+					Repo:        repo,
+					Directory:   directory,
+					Branch:      nil,
+					Hostname:    nil,
+					APIEndpoint: nil,
+				},
+				UpdateSubdependencies: false,
+				UpdatingAPullRequest:  false,
+			},
+		}
+	}
+
+	if inputServerPort != 0 {
+		input = server.Input(inputServerPort)
```

Comment:
This surprised me... if there's input over the network, it takes precedence over the command line input? I'd have expected it to throw an error saying that it got ambiguous input of both network and cmd flags and to only specify one or the other.